src/display/editor/tools.js

```diff
@@ -1469,6 +1471,89 @@ class AnnotationEditorUIManager {
     }
   }
 
+  setUpDragSession() {
+    if (!this.hasSelection) {
+      return;
+    }
+    // Avoid to have spurious text selection in the text layer when dragging.
+    this.disableUserSelect(true);
+    this.#draggingEditors = new Map();
+    for (const editor of this.#selectedEditors) {
+      this.#draggingEditors.set(editor, [
+        editor.x,
+        editor.y,
+        editor.parent,
+        null,
+        null,
+        null,
+      ]);
+    }
+  }
+
+  /**
+   * Ends the drag session.
+   * @returns {boolean} true if at least one editor has been moved.
+   */
+  endDragSession() {
+    if (!this.#draggingEditors) {
+      return false;
+    }
+    this.disableUserSelect(false);
+    const map = this.#draggingEditors;
+    this.#draggingEditors = null;
+    let mustBeAddedInUndoStack = false;
+
+    for (const [{ x, y, parent }, value] of map) {
+      value[3] = x;
+      value[4] = y;
+      value[5] = parent;
+      mustBeAddedInUndoStack ||=
+        x !== value[0] || y !== value[1] || parent !== value[2];
+    }
+
+    if (!mustBeAddedInUndoStack) {
+      return false;
+    }
+    this.addCommands({
+      cmd: () => {
+        for (const [editor, [, , , newX, newY, newParent]] of map) {
+          if (!this.#allEditors.has(editor.id)) {
+            continue;
+          }
+          newParent.changeParent(editor);
+          editor.x = newX;
+          editor.y = newY;
+          editor.fixAndSetPosition();
+          editor.moveInDOM();
+        }
+      },
+      undo: () => {
+        for (const [editor, [savedX, savedY, savedParent]] of map) {
+          if (!this.#allEditors.has(editor.id)) {
+            continue;
+          }
+          savedParent.changeParent(editor);
+          editor.x = savedX;
+          editor.y = savedY;
+          editor.fixAndSetPosition();
+          editor.moveInDOM();
```

Comment:
With the helper function mentioned above this could become:
```suggestion
          if (this.#allEditors.has(editor.id)) {
            move(editor, savedParent, savedX, savedY);
          }
```